src/pages/tBTC/Bridge/components/TimelineItem.tsx

```diff
@@ -1,45 +1,315 @@
-  imageSrc?: any
-  withBadge?: boolean
-} & StepProps
-
-const TimelineItem: FC<TimelineProps> = ({
-  stepText,
-  helperLabelText,
+} & AccordionItemProps
+
+const StyledAccordionIcon: FC<{ isExpanded: boolean }> = ({ isExpanded }) => (
+  <AccordionIcon
+    w={4}
+    h={4}
+    color="hsl(182, 100%, 70%)"
+    transform={`rotate(${isExpanded ? -90 : 0}deg)`}
+  />
+)
+
+type BulletSymbolStatus = "completed" | "active" | "incoming"
+const BulletSymbol: FC<{ status?: BulletSymbolStatus } & BoxProps> = ({
+  status = "incoming",
+}) => {
+  const baseColors: [BulletSymbolStatus, string][] = [
+    ["completed", "hsl(151, 100%, 70%)"],
+    ["active", "hsl(182, 100%, 70%)"],
+  ]
+  const backgroundColor = new Map<BulletSymbolStatus, string>([
+    ...baseColors,
+    ["incoming", "transparent"],
+  ])
+
+  const outlineColor = new Map<BulletSymbolStatus, string>([
+    ...baseColors,
+    ["incoming", "hsl(0, 0%, 20%)"],
+  ])
+
+  return (
+    <Box
+      rounded="full"
+      w={status !== "completed" ? 3 : 6}
+      h={status !== "completed" ? 3 : 6}
+      bgColor={backgroundColor.get(status)}
+      outline={
+        status !== "completed"
+          ? `1px solid ${outlineColor.get(status)}`
+          : undefined
+      }
+      outlineOffset={"5px" /* 6px - 1px of outline thickness */}
+    >
+      {status === "completed" ? (
+        <Icon
+          as={CompleteIcon}
+          w="full"
+          h="full"
+          p={1.5}
+          color="hsl(0, 0%, 7%)"
+        />
+      ) : null}
+    </Box>
+  )
+}
+
+const PrimaryTimelineItem: FC<Omit<TimelineItemProps, "variant">> = ({
+  number,
+  label,
+  isComplete,
+  isActive,
+  children,
+  ...restProps
+}) => {
+  return (
+    <AccordionItem
+      {...restProps}
+      mb={0}
+      borderWidth={0}
+      _last={{
+        borderWidth: 0,
+        "> div": {
+          mb: 0,
+        },
+      }}
+    >
+      {({ isExpanded }) => (
+        <Box pb={isExpanded ? 0 : 2}>
+          <VStack spacing={0}>
+            <HStack
+              as={AccordionButton}
+              spacing={2.5}
+              p={0}
+              pointerEvents={"unset"}
+              _hover={{ bg: "transparent" }}
+            >
+              <StyledAccordionIcon isExpanded={isExpanded} />
+              <HStack justify="space-between" w="full">
+                <BodyMd color={"white"}>{label}</BodyMd>
+              </HStack>
+            </HStack>
+            <AccordionPanel
+              pt={0}
+              pr={0}
+              pb={8}
+              pl={16 + 10 /* icon width + spacing in pixels */}
+              mt={2}
+              mb={0}
+              ml={0}
+            >
+              <BodySm color="hsla(0, 0%, 100%, 50%)" fontWeight="medium">
+                {children}
+              </BodySm>
+            </AccordionPanel>
+          </VStack>
+        </Box>
+      )}
+    </AccordionItem>
+  )
+}
+const SecondaryTimelineItem: FC<Omit<TimelineItemProps, "variant">> = ({
```

Comment:
Ahh, I see. In that case I propose to name the variants better, because right now it's hard to know what each variant does without looking deep into the code.

My proposition:
1. "simple" - this will be the "primary" one. It will indicate that this is the simplest fom of the timeline without any nuances.
2. "numbered" - this will be the "secondary" one. It will indicate that this timeline variant is numbered
3. "pulse" - this will be the "tertiary" one. This name captures the dynamic, changing nature of the dot indicator based on completion, suggesting a lively or active status change, much like a pulse fluctuates.

What do you think?

Reply:
Second thing is that the expandable/no-expandable feature should not be tied with variant IMO. For the `src/component/Timeline.jsx` component I would add additional property such as `expandOnClick: boolean` and pass it down to children (`TimelineItem`).

Then, in `MintingDepositTimeline`, we should pass `expandOnClick={true}` for `simple` (AKA `primary`) variant and `expandOnClik={false}` for `numbered` (AKA `numbered`) and `pulse` (AKA `tertiary`) variants

Reply:
It's impossible to implement it that way in current form. It requires to make the logic of toggling items programmatically embeddable which is impossible because it depends on the specific variables and it's values (`mintingStep`/`depositStep`). In my opinion it's okay to have it that way. That's what this variants actually are for. 
To achieve what you propose I might need to implement a universal way to provide flexible source of truth for currently expanded item. It's doable but probably too much time consuming at this point. Let's keep it in backlog and eventually implement it in future however to be honest I don't see significant need of it. Coupling it with specific variant is enough.

Reply:
Ref commit: fe0f1479

Reply:
Summing up:

The `secondary` variant of the `Timeline` was not wroking properly outside of `MintingDepositTimeline` component, because it was not possible to click and expand inactive items. Turned out the `pointerEvents` were set to "none" for this variant resulting in this unexpected behaviour. It was fixed in 914d15fcf2d9cd6c63a733a27304c5d6ddcc57dd